### PR TITLE
feat: telegram notifications

### PR DESCRIPTION
**Pull request by kolaente**
This PR adds support for telegram notifications if the status of a monitor changes.

### TODO

* [x] Docs

### CODE REVIEW

**Comment by kolaente** [comment]
I currently have the problem that the messages are not arriving properly and I don't quite know how to debug this. I figured I can see web requests with `wrangler tail` but that does not show me the output of triggered jobs. Any ideas?

**Comment by eidam** [comment]
@kolaente there is an API path I am using locally for developing the triggered jobs, that might help you

https://github.com/eidam/cf-workers-status-page/blob/main/pages/api/triggerCron.js#L5

**Comment by kolaente** [comment]
That looks promising, thanks. I'll see if I can find something with that.

File: src/functions/helpers.js [hunk 1]
```diff
@@ -54,6 +56,23 @@ export async function notifySlack(monitor, operational) {
   })
 }
 
+export async function notifyTelegram(monitor, operational) {
+  const text = `${operational ? '✅' : '❌'}
+  \`${monitor.method ? monitor.method : "GET"} ${monitor.url}\` - 👀 <${config.settings.url}|Status Page>
+  Monitor *${monitor.name}* changed status to *${getOperationalLabel(operational)}*`
+
+  const payload = new FormData()
+  payload.append('chat_id', TELEGRAM_CHAT_ID)
```
Comment (eidam) [inline src/functions/helpers.js:65]: `SECRET_` prefix is missing in your telegram secrets

Comment (kolaente) [reply src/functions/helpers.js]: I guess that explains a lot.

Comment (kolaente) [reply src/functions/helpers.js]: Fixed

**Comment by kolaente** [comment]
@eidam How do I use the KV storage when testing locally? I currently get 
```
ReferenceError: KV_STATUS_PAGE is not defined
```
using `wrangler dev`

**Comment by eidam** [comment]
Create test namespace (or use the same if you don't mind) and add the following to the wrangler.toml, it will create the binding for your worker (this is done automatically in GitHub Actions)
```
kv-namespaces = [{binding="KV_STATUS_PAGE", id="xxx", preview_id="xxx"}]
```

**Comment by kolaente** [comment]
That seems to work, thanks!

**Comment by kolaente** [comment]
Looks like the issue is this with telegram:

```json
{
  "description": "Bad Request: can't parse entities: Character '-' is reserved and must be escaped with the preceding '\\'",
  "error_code": 400,
  "ok": false
}
```

Not sure why `-` is a reserved character but hey...

It works if I omit evey `-`. Also this means I can't have the url in the message.

**Comment by kolaente** [comment]
On the other hand, this also means people can't use `-` in their monitor names when using telegram notifications - I guess it would be better to just escape that?

**Comment by eidam** [comment]
In that case, I think we should escape full message, gonna check the telegram API docs

btw, I just merged some bigger refactor to `develop` branch, would you mind to open the PR against that one? There will be some conflicts, would be great if you could do that - but I will do it for you otherwise later today :)

**Comment by kolaente** [comment]
> In that case, I think we should escape full message, gonna check the telegram API docs

I've added a very rough escaping of the name - the rest works since it's in "`".

> btw, I just merged some bigger refactor to develop branch, would you mind to open the PR against that one? 

Will do :)

**Comment by kolaente** [comment]
Now I don't get any status updates at all, I guess that's because of the changes in `develop`?

File: src/functions/cronTrigger.js [hunk 2]
```diff
@@ -86,6 +87,12 @@ export async function processCronTrigger(event) {
         res: {},
       }
     }
+    // Send Telegram message on monitor change
+    if (monitorsState[monitor.id].operational !== monitorOperational
```
Comment (eidam) [inline src/functions/cronTrigger.js:91]:
```suggestion
    if (monitorStatusChanged
```

Comment (eidam) [reply src/functions/cronTrigger.js:91]: this should resolve the issue after changes in develop :)

Comment (kolaente) [reply src/functions/cronTrigger.js]: Ohh I see, thanks!

**Comment by eidam** [comment]
@kolaente looks nice! if ready for merge, please clean up the config, remove the branch from `deploy.yml` triggers and squash the commits

**Comment by kolaente** [comment]
Looks like I messed up that rebase -.-

**Comment by kolaente** [comment]
There we go!

@eidam All good?

**Review by eidam — APPROVED** [review]
looks great, many thanks!

**Comment by kolaente** [comment]
Thanks for merging!

**Comment by eidam** [comment]
thank you for the contribution! :)

**Pull request merged**